mitodl/mit-open
# Add accessibility linting

### What are the relevant tickets?
Closes https://github.com/mitodl/hq/issues/5155 (for now; more could probably be done).

### Description (What does it do?)
This PR adds [eslint-plugin-styled-components-a11y](https://www.npmjs.com/package/eslint-plugin-styled-components-a11y?activeTab=readme) to lint accessibility issues. This package is a wrapper around [eslint-plugin-jsx-a11y](https://www.npmjs.com/package/eslint-plugin-jsx-a11y).

### How can this be tested?
- Linting should pass
- Try removing `aria-label` from an ActionButton; it should give you a useful error message
- Try removing an alt tag from an `img`, or any `styled.img`. You should get a useful error message.

### Additional Context
One of the main user-facing changes here is that I set many alt tags to `""` (e.g., on testimonials). This is better than no alt tag, and IMO is appropriate. These images are generally decorative, not, e.g., diagrams or a literal, physical product. And the testimonial APIs don't provide alt data, so we don't really have another choice.
